### ISSUE TITLE
Finetuning uses internet access for some reason?

### ISSUE THREAD

**Issue by Urammar** [erew123/alltalk_tts#56]
![image](https://github.com/erew123/alltalk_tts/assets/40313782/6e97f52d-805e-4b6f-a0b3-caf38a376bae)

Finetuning models is periodically sending bulk network traffic, and actually errors out if the internet is disconnected when instigating training. Why would local finetuning require an internet connection?

Additionally, training writes to disk very frequently, I have 64gb of ram, requesting that be uses for checkpoints, and a final, singular, write at end of training be done instead of thrashing my ssd.

**Comment by erew123** [comment]
The only thing I know about was that the Coqui training would collect telemetry on training, however I disabled that [line 32 here](https://github.com/erew123/alltalk_tts/blob/main/finetune.py). And I know Gradio tries to set up a shared interface, but that should also be disabled. 

Which step is this on that you are seeing the traffic? Or are you just saying generally it sends it?

**Comment by Urammar** [comment]
Then that is very strange.

This is on the training step, it was actively training during this screenshot. And as you said, the gradio share was not enabled.

**Comment by erew123** [comment]
Ill take a look at some point and double check its applying the setting I specified... but that's definitely the setting (according to Coqui) https://github.com/coqui-ai/Trainer?tab=readme-ov-file#anonymized-telemetry

There isnt really anything else going on at that stage that should send out anything else.

**Comment by erew123** [comment]
I've looked into this and tested as much as I can. Due to the way some OS's handle environment variables passed through Python, there is no way to universally ensure this is disabled, bar setting it pre starting finetune.py. With the environment variable set you are fully able to finetune offline. I've detailed this here https://github.com/erew123/alltalk_tts/tree/main#-a-note-about-anonymous-training-telemetry-information--disabling-it

Hope that answers everything fully. I'm going to close this ticket off. 

Thanks
